design/src/rfcs/rfc0022_refine_builder.md

```diff
@@ -0,0 +1,556 @@
+
+Services with a high number of routes can lead to fairly long startup routines.  
+Developers might be tempted to break down the startup routine into smaller functions, grouping together operations with common requirements (similar domain, same middlewares, etc.).
+
+What does the signature of those smaller functions look like?  
+The service builder must be one of the arguments if we want to register handlers. We must also return it to allow the orchestrating function to finish the application setup (our setters take ownership of `self`).
+
+A first sketch:
+
+```rust
+fn partial_setup(builder: PokemonServiceBuilder) -> PokemonServiceBuilder {
+    /* */
+}
+```
+
+The compiler demands to see those generic parameters in the signature:
+
+```text
+error[E0107]: missing generics for struct `PokemonServiceBuilder`
+  --> pokemon-service/src/bin/pokemon-service.rs:28:27
+   |
+28 | fn partial_setup(builder: PokemonServiceBuilder) -> PokemonServiceBuilder {
+   |                           ^^^^^^^^^^^^^^^^^^^^^ expected at least 6 generic arguments
+   |
+note: struct defined here, with at least 6 generic parameters: `Op1`, `Op2`, `Op3`, `Op4`, `Op5`, `Op6`
+
+error[E0107]: missing generics for struct `PokemonServiceBuilder`
+  --> pokemon-service/src/bin/pokemon-service.rs:28:53
+   |
+28 | fn partial_setup(builder: PokemonServiceBuilder) -> PokemonServiceBuilder {
+   |                                                     ^^^^^^^^^^^^^^^^^^^^^ expected at least 6 generic arguments
+   |
+note: struct defined here, with at least 6 generic parameters: `Op1`, `Op2`, `Op3`, `Op4`, `Op5`, `Op6`
+```
+
+We could try to nudge the compiler into inferring them:
+
+```rust
+fn partial_setup(
+    builder: PokemonServiceBuilder<_, _, _, _, _, _>,
+) -> PokemonServiceBuilder<_, _, _, _, _, _> {
+    /* */
+}
+```
+
+but that won't fly either:
+
+```text
+error[E0121]: the placeholder `_` is not allowed within types on item signatures for return types
+  --> pokemon-service/src/bin/pokemon-service.rs:30:28
+   |
+30 | ) -> PokemonServiceBuilder<_, _, _, _, _, _> {
+   |                            ^  ^  ^  ^  ^  ^ not allowed in type signatures
+   |                            |  |  |  |  |
+   |                            |  |  |  |  not allowed in type signatures
+   |                            |  |  |  not allowed in type signatures
+   |                            |  |  not allowed in type signatures
+   |                            |  not allowed in type signatures
+   |                            not allowed in type signatures
+```
+
+We must type it all out:
+
+```rust
+fn partial_setup<Op1, Op2, Op3, Op4, Op5, Op6>(
+    builder: PokemonServiceBuilder<Op1, Op2, Op3, Op4, Op5, Op6>,
+) -> PokemonServiceBuilder<Op1, Op2, Op3, Op4, Op5, Op6> {
+    builder
+}
+```
+
+That compiles, at last.  
+Let's try to register an operation handler now:
+
+```rust
+fn partial_setup<Op1, Op2, Op3, Op4, Op5, Op6>(
```

Comment:
Far from ideal, but for simple use cases, you could get away with a macro that performs this initialization.